src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp

```diff
@@ -21,6 +21,138 @@ constexpr char biosIntType[] =
+            return;
+        }
+        log<level::INFO>("BaseBIOSTable - property changed");
+
+        // Check if the IP address has changed (i.e., if current ip address in
+        // the biosTableAttrs data member and ip address in bios table are
+        // different)
+        auto biosTableAttrs = manager.getBIOSTableAttrs();
+        for (auto i : biosTableAttrs)
+        {
+            if (i.first == "vmi_if0_ipv4_ipaddr" ||
+                i.first == "vmi_if1_ipv4_ipaddr")
+            {
+                std::string currIpAddr = std::get<std::string>(i.second);
+                if (currIpAddr.empty())
+                {
+                    log<level::INFO>("Current IP in biosAttrs copy is empty");
+                    return;
+                }
+
+                std::string ipAddr = getAttrFromBiosTable(i.first);
+                if (ipAddr != currIpAddr)
+                {
+                    // Ip address has been updated. Check if it is dhcp.
+                    // This method was intended to watch the bios table
+                    // property change signal and restart the hypervisor
+                    // networkd service whenever the dhcp server has provided an
+                    // IP from different range. Because, in all other cases,
+                    // user configures ip that will be set in the dbus object,
+                    // followed by bios table updation. Only in this dhcp case,
+                    // the dbus will not be having the updated ip address which
+                    // is in bios table. This method is to sync the ip addresses
+                    // between the bios table & dbus object.
+                    std::string intf;
+                    if (i.first == "vmi_if0_ipv4_ipaddr")
+                    {
+                        intf = "if0";
+                    }
+                    else if (i.first == "vmi_if1_ipv4_ipaddr")
+                    {
+                        intf = "if1";
+                    }
+
+                    std::string dhcpEnabled =
+                        getAttrFromBiosTable("vmi_" + intf + "_ipv4_method");
+                    if (dhcpEnabled == "IPv4DHCP")
+                    {
+                        log<level::INFO>("DHCP enabled. Restarting hypervisor "
+                                         "networkd service.");
+                        std::string unit =
+                            "xyz.openbmc_project.Network.Hypervisor.service";
+                        try
+                        {
+                            auto method = bus.new_method_call(
+                                "org.freedesktop.systemd1",
+                                "/org/freedesktop/systemd1",
+                                "org.freedesktop.systemd1.Manager",
+                                "RestartUnit");
+                            method.append(unit.c_str(), "replace");
+                            bus.call_noreply(method);
+                        }
```

Comment:
i am not sure why do you need to restart the hypervisor service , you can just update the dbus objects with new values..restarting this service might lead to tmp 404 status for HMC.

Reply:
Okay, this was because, every time I update any dbus properties, in its setter method I am setting the same value in the biostable as well.
So now if I update the dbus obj values, it means that I am calling the setter method of those ip address properties. what happens is:
call setter method address(new ip) => sets the dbus and updates the biostable attribute with new ip (but it will already have the newip, so this update would be redundant)

May be i should be checking the bios table attribute before setting it - if it is same, leave it as such, else go ahead and update it!

Reply:
Done